File: week6/pset6/sentimental-readability/readability.py
from cs50 import get_string

file = get_string("Text: ")
W = 0
L = 0
S = 0

for i in range(len(file)):
    if file[i].isspace() == True:
        W += 1
    if file[i].isalpha() == True:
        L += 1
    if file[i] in ['?', '.', '!']:
        S += 1
W += 1
L = L / W * 100
S = S / W * 100

# L is the average number of letters per 100 words in the text
# S is the average number of sentences per 100 words in the text.
# print(W, L, S)

index = round(0.0588 * L - 0.296 * S - 15.8)
if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: week2/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string text = get_string("Text: ");

    int letter = 0;
    int words = 0;
    int sentence = 0;
    for (int i = 0, L = strlen(text); i<L; i++)
    {
        if (isalpha(text[i]))
        {
            letter ++;
        }

        if (isspace(text[i]))
        //利用空格来检测,空格数量+1
        //但是要确保won't start or end with a word
        {
            words ++;
        }
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentence ++;
        }
    }
    float S = ((float)sentence / (words + 1)) * 100.0;
    float L = ((float)letter / (words + 1)) * 100.0;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}